### PR TITLE
chore: set up commit hooks

### LINKED ISSUE #26: Automate commit hooks
Commit message conventions could be important for releases.

Possible solutions:
- [commitlint](https://github.com/conventional-changelog/commitlint#readme)
- [commitizen](https://github.com/commitizen/cz-cli#readme)
- [husky](https://typicode.github.io/husky/)

What to include:
- [x] message hook
  - [x] Commitlint
    - [x] scopes are derived from Nx projects 
    - [x] conventional commit
- [x] pre-commit hook
  - [x] formatting with Prettier
  - [x] Lint in CI

### PR DESCRIPTION
**Pull request by matejchalk**
Closes #26 

## Git hook (Commitlint)

image

## Commiting with prompts (Commitizen)

image

image

### CODE REVIEW

**Review by vmasek — DISMISSED** [review]

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { RuleConfigSeverity } = require('@commitlint/types');
+const {
+  utils: { getProjects },
+} = require('@commitlint/config-nx-scopes');
+
+/** @type {import('@commitlint/types').UserConfig} */
+const configuration = {
+  extends: ['@commitlint/config-conventional'],
+  rules: {
+    'scope-enum': async ctx => {
```
Comment (BioPhoton) [inline commitlint.config.js:10]: Should we consider a scope for root level and general things like GH actions etc?

**Review by BioPhoton — COMMENTED** [review]
Nice quality of life improvement!!

File: commitlint.config.js (hunk 1 above)
Comment (BioPhoton) [reply commitlint.config.js:10]: Sounds good!

scopes:
- `tools`
- `workflows`
- `testing` => for generic testing helper?

Comment (matejchalk) [reply commitlint.config.js:10]: Done :heavy_check_mark:

**Review by BioPhoton — APPROVED** [review]

**Pull request merged**